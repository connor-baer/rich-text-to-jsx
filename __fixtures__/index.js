/* eslint-disable max-len */
export function createDocument(content) {
  return {
    data: {},
    nodeType: 'document',
    content
  };
}

export const headingOne = {
  data: {},
  content: [
    {
      data: {},
      marks: [],
      value: 'This is a first-level heading',
      nodeType: 'text'
    }
  ],
  nodeType: 'heading-1'
};

export const headingThree = {
  data: {},
  content: [
    {
      data: {},
      marks: [],
      value: 'This is a third-level heading',
      nodeType: 'text'
    }
  ],
  nodeType: 'heading-3'
};

export const text = {
  data: {},
  marks: [],
  value: 'This is normal text.',
  nodeType: 'text'
};

export const bold = {
  data: {},
  marks: [
    {
      type: 'bold'
    }
  ],
  value: 'This is bold text.',
  nodeType: 'text'
};

export const italic = {
  data: {},
  marks: [
    {
      type: 'italic'
    }
  ],
  value: 'This is italic text.',
  nodeType: 'text'
};

export const underline = {
  data: {},
  marks: [
    {
      type: 'underline'
    }
  ],
  value: 'This is underlined text.',
  nodeType: 'text'
};

export const hyperlink = {
  data: {
    uri: 'https://acme.com'
  },
  content: [
    {
      data: {},
      marks: [],
      value: 'This is a hyperlink.',
      nodeType: 'text'
    }
  ],
  nodeType: 'hyperlink'
};

export const boldAndItalic = {
  data: {},
  marks: [
    {
      type: 'bold'
    },
    {
      type: 'italic'
    }
  ],
  value: 'This is bold and italic text.',
  nodeType: 'text'
};

export const paragraph = {
  data: {},
  content: [text, bold, italic, underline, hyperlink],
  nodeType: 'paragraph'
};

export const blockquote = {
  data: {},
  content: [paragraph],
  nodeType: 'blockquote'
};

export const listItem = {
  data: {},
  content: [paragraph],
  nodeType: 'list-item'
};

export const unorderedList = {
  data: {},
  content: [listItem, listItem, listItem],
  nodeType: 'unordered-list'
};

export const hr = {
  data: {},
  content: [],
  nodeType: 'hr'
};

export const image = {
  id: '4fgGUXCJXWOQUAEQqCS8MW',
  updatedAt: '2019-01-14T04:57:27.049Z',
  title: 'Random photo from Unsplash.com',
  file: {
    url: 'https://source.unsplash.com/random/800x500',
    details: {
      size: 240963,
      image: {
        width: 800,
        height: 500
      }
    },
    fileName: 'random-unsplash.jpg',
    contentType: 'image/jpg'
  }
};

export const video = {
  id: '4fgGUXCJXWOQUAEQqCS8MW',
  updatedAt: '2019-01-14T04:57:27.049Z',
  title: 'Example video from YouTube',
  file: {
    url: 'https://youtube.com/example.mp4',
    details: {
      size: 2409638,
      video: {
        width: 1920,
        height: 1080
      }
    },
    fileName: 'example.mp4',
    contentType: 'video/mp4'
  }
};

export const audio = {
  id: '4fgGUXCJXWOQUAEQqCS8MW',
  updatedAt: '2019-01-14T04:57:27.049Z',
  title: 'Example audio from Spotify',
  file: {
    url: 'https://spotify.com/example.mp3',
    details: {
      size: 24096
    },
    fileName: 'example.mp3',
    contentType: 'audio/mp3'
  }
};

export const embeddedImage = {
  data: {
    target: image
  },
  content: [],
  nodeType: 'embedded-asset-block'
};

export const embeddedVideo = {
  data: {
    target: video
  },
  content: [],
  nodeType: 'embedded-asset-block'
};

export const embeddedAudio = {
  data: {
    target: audio
  },
  content: [],
  nodeType: 'embedded-asset-block'
};

export const assetHyperlink = {
  data: {
    target: image
  },
  content: [
    {
      data: {},
      marks: [],
      value: 'ham hock',
      nodeType: 'text'
    }
  ],
  nodeType: 'asset-hyperlink'
};

const entry = {
  id: '32v7TZ7YQEaugOeew4SymY',
  contentType: 'page',
  updatedAt: '2019-01-18T14:37:30.221Z',
  slug: 'about',
  title: 'Title: About Acme Corp'
};

export const entryHyperlink = {
  data: {
    target: {
      id: '32v7TZ7YQEaugOeew4SymY',
      contentType: 'route',
      updatedAt: '2019-01-18T14:37:30.221Z',
      slug: 'about'
    }
  },
  content: [
    {
      data: {},
      marks: [],
      value: 'This is a link to an entry.',
      nodeType: 'text'
    }
  ],
  nodeType: 'entry-hyperlink'
};

export const embeddedEntryInline = {
  data: {
    target: entry
  },
  content: [],
  nodeType: 'embedded-entry-inline'
};

export const embeddedEntryBlock = {
  data: {
    target: entry
  },
  content: [],
  nodeType: 'embedded-entry-block'
};
