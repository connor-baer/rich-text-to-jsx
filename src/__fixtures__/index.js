import { BLOCKS, INLINES, MARKS } from '@contentful/rich-text-types';

/* eslint-disable max-len */
export function createDocument(content) {
  return {
    data: {},
    nodeType: 'document',
    content,
  };
}

export const headingOne = {
  data: {},
  content: [
    {
      data: {},
      marks: [],
      value: 'This is a first-level heading',
      nodeType: 'text',
    },
  ],
  nodeType: BLOCKS.HEADING_1,
};

export const headingThree = {
  data: {},
  content: [
    {
      data: {},
      marks: [],
      value: 'This is a third-level heading',
      nodeType: 'text',
    },
  ],
  nodeType: BLOCKS.HEADING_3,
};

export const text = {
  data: {},
  marks: [],
  value: 'This is normal text.',
  nodeType: 'text',
};

export const bold = {
  data: {},
  marks: [
    {
      type: MARKS.BOLD,
    },
  ],
  value: 'This is bold text.',
  nodeType: 'text',
};

export const italic = {
  data: {},
  marks: [
    {
      type: MARKS.ITALIC,
    },
  ],
  value: 'This is italic text.',
  nodeType: 'text',
};

export const underline = {
  data: {},
  marks: [
    {
      type: MARKS.UNDERLINE,
    },
  ],
  value: 'This is underlined text.',
  nodeType: 'text',
};

export const hyperlink = {
  data: {
    uri: 'https://acme.com',
  },
  content: [
    {
      data: {},
      marks: [],
      value: 'This is a hyperlink.',
      nodeType: 'text',
    },
  ],
  nodeType: INLINES.HYPERLINK,
};

export const boldAndItalic = {
  data: {},
  marks: [
    {
      type: MARKS.BOLD,
    },
    {
      type: MARKS.ITALIC,
    },
  ],
  value: 'This is bold and italic text.',
  nodeType: 'text',
};

export const paragraph = {
  data: {},
  content: [text, bold, italic, underline, hyperlink],
  nodeType: BLOCKS.PARAGRAPH,
};

export const blockquote = {
  data: {},
  content: [paragraph],
  nodeType: BLOCKS.QUOTE,
};

export const listItem = {
  data: {},
  content: [paragraph],
  nodeType: BLOCKS.LIST_ITEM,
};

export const unorderedList = {
  data: {},
  content: [listItem, listItem, listItem],
  nodeType: BLOCKS.UL_LIST,
};

export const hr = {
  data: {},
  content: [],
  nodeType: BLOCKS.HR,
};

export const assetLink = {
  sys: {
    id: '9mpxT4zsRi6Iwukey8KeM',
    link: 'Link',
    type: 'Asset',
  },
};

export const image = {
  sys: {
    id: '4fgGUXCJXWOQUAEQqCS8MW',
    updatedAt: '2019-01-14T04:57:27.049Z',
  },
  fields: {
    title: 'Random photo from Unsplash.com',
    file: {
      url: 'https://source.unsplash.com/random/800x500',
      details: {
        size: 240963,
        image: {
          width: 800,
          height: 500,
        },
      },
      fileName: 'random-unsplash.jpg',
      contentType: 'image/jpg',
    },
  },
};

export const video = {
  sys: {
    id: '4fgGUXCJXWOQUAEQqCS8MW',
    updatedAt: '2019-01-14T04:57:27.049Z',
  },
  fields: {
    title: 'Example video from YouTube',
    file: {
      url: 'https://youtube.com/example.mp4',
      details: {
        size: 2409638,
        video: {
          width: 1920,
          height: 1080,
        },
      },
      fileName: 'example.mp4',
      contentType: 'video/mp4',
    },
  },
};

export const audio = {
  sys: {
    id: '4fgGUXCJXWOQUAEQqCS8MW',
    updatedAt: '2019-01-14T04:57:27.049Z',
  },
  fields: {
    title: 'Example audio from Spotify',
    file: {
      url: 'https://spotify.com/example.mp3',
      details: {
        size: 24096,
      },
      fileName: 'example.mp3',
      contentType: 'audio/mp3',
    },
  },
};

export const embeddedImage = {
  data: {
    target: image,
  },
  content: [],
  nodeType: BLOCKS.EMBEDDED_ASSET,
};

export const embeddedVideo = {
  data: {
    target: video,
  },
  content: [],
  nodeType: BLOCKS.EMBEDDED_ASSET,
};

export const embeddedAudio = {
  data: {
    target: audio,
  },
  content: [],
  nodeType: BLOCKS.EMBEDDED_ASSET,
};

export const assetHyperlink = {
  data: {
    target: image,
  },
  content: [
    {
      data: {},
      marks: [],
      value: 'ham hock',
      nodeType: 'text',
    },
  ],
  nodeType: INLINES.ASSET_HYPERLINK,
};

export const entryLink = {
  sys: {
    id: '9mpxT4zsRi6Iwukey8KeM',
    link: 'Link',
    linkType: 'Entry',
  },
};

const entry = {
  sys: {
    id: '32v7TZ7YQEaugOeew4SymY',
    contentType: { sys: { id: 'page' } },
    updatedAt: '2019-01-18T14:37:30.221Z',
  },
  fields: {
    slug: 'about',
    title: 'Title: About Acme Corp',
  },
};

export const entryHyperlink = {
  data: {
    target: entry,
  },
  content: [
    {
      data: {},
      marks: [],
      value: 'This is a link to an entry.',
      nodeType: 'text',
    },
  ],
  nodeType: INLINES.ENTRY_HYPERLINK,
};

export const embeddedEntryInline = {
  data: {
    target: entry,
  },
  content: [],
  nodeType: INLINES.EMBEDDED_ENTRY,
};

export const embeddedEntryBlock = {
  data: {
    target: entry,
  },
  content: [],
  nodeType: BLOCKS.EMBEDDED_ENTRY,
};
