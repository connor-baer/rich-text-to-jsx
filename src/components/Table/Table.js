import React from 'react';

/**
 * Default element for tables.
 */
function Table({ children }) {
  return (
    <table>
      <tbody>{children}</tbody>
    </table>
  );
}

if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  const PropTypes = require('prop-types');

  Table.propTypes = {
    /**
     * The table rows and cells
     */
    children: PropTypes.node,
  };
}

/**
 * @component
 */
export default Table;
