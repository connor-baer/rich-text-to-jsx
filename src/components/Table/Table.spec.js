import React from 'react';
import { create } from 'react-test-renderer';

import { table } from '../../__fixtures__';

import Table from './Table';

describe('Table', () => {
  it('should render a table', () => {
    const actual = create(<Table {...table} />);
    expect(actual).toMatchSnapshot();
  });
});
