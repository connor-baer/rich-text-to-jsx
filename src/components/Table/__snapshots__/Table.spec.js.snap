// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Table should render a table 1`] = `
<table>
  <tbody />
</table>
`;
